distributed/core.py

```diff
@@ -859,13 +859,30 @@ async def listen(self, port_or_addr=None, allow_offload=True, **kwargs):
         else:
             addr = port_or_addr
             assert isinstance(addr, str)
-        listener = await listen(
-            addr,
-            self.handle_comm,
-            deserialize=self.deserialize,
-            allow_offload=allow_offload,
-            **kwargs,
-        )
+        try:
+            listener = await listen(
+                addr,
+                self.handle_comm,
+                deserialize=self.deserialize,
+                allow_offload=allow_offload,
+                **kwargs,
+            )
+        except OSError:
+            fallback_port_or_addr = kwargs.get("fallback_port_or_addr", None)
+            if not fallback_port_or_addr:
+                raise
+            warnings.warn(
```

Comment:
this line is what causes the test `distributed/deploy/tests/test_local.py::test_Client_twice` to fail, if I just comment the warning, it passes. Should I remove the warning, change it to `logger` invocation instead, or is there some means of making the test accept a warning occurrence?